src/DBmysqlIterator.php

```diff
@@ -113,7 +113,14 @@ public function __construct($dbconnexion)
     public function execute($table, $crit = "", $debug = false)
     {
         $this->buildQuery($table, $crit, $debug);
-        $this->res = ($this->conn ? $this->conn->query($this->sql) : false);
+        $this->res = false;
+        if ($this->conn) {
+            if ($this->conn instanceof DBmysql) {
+                $this->res = $this->conn->doQuery($this->sql);
+            } else {
+                $this->res = $this->conn->query($this->sql);
+            }
```

Comment:
Is there any reason for this fallback? I do not find any case where `$this->conn` would not be an instance of `DBmysql`.